File: server/api/v1/autocode/servicefields.go
package autocode

import (
	"encoding/json"
	"fmt"

	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/model/autocode"
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/response"

	"github.com/gin-gonic/gin"
	// "go.uber.org/zap"
)

type ServiceFieldsApi struct {
}

// var servicefieldsService = service.ServiceGroupApp.AutoCodeServiceGroup.FieldsService

// CreateServiceFields 创建Fields
// @Tags ServiceFields
// @Summary 创建ServiceFields
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body autocode.ServiceFields true "创建Fields"
// @Success 200 {string} string "{"success":true,"data":{},"msg":global.Translate("general.getDataSuccess")}"
// @Router /servicefields/CreateServiceFields [post]
func (fieldsApi *ServiceFieldsApi) CreateServiceFields(c *gin.Context) {
	var fields autocode.ServiceFields
	_ = c.ShouldBindJSON(&fields)

	byteValueReq, err := json.Marshal(fields)
	fmt.Println(string(byteValueReq))

	if err != nil {
		response.FailWithMessage(global.Translate("general.creationFailErr"), c)

	}

	url := global.GVA_VP.GetString("gateway-controller.url") + "/fields/add"
	fmt.Println(url)
	body, err := global.SendPostReq("POST", byteValueReq, url)
	fmt.Println(string(body))

	if err != nil {

		response.FailWithMessage(global.Translate("general.creationFailErr"), c)
	} else {

		response.OkWithMessage(global.Translate("general.success"), c)
	}

}

File: web/src/api/servicefields.js
import service from '@/utils/request'

// @Tags Fields
// @Summary 创建Fields
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Fields true "创建Fields"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /servicefields/createServiceFields [post]
export const createServiceFields = (data) => {

    return service({
        url: '/servicefields/createServiceFields',
        method: 'post',
        data
    })

}

export const updateServiceFields = (data) => {

    return service({
        url: '/servicefields/updateServiceFields',
        method: 'put',
        data
    })

}

export const deleteServiceFields = (data) => {

    return service({
        url: '/servicefields/deleteServiceFields',
        method: 'delete',
        data
    })

}